spring-integration-core/src/main/java/org/springframework/integration/endpoint/AbstractEndpoint.java

```diff
@@ -84,6 +108,21 @@ protected void onInit() throws Exception {
 				}
 			}
 		}
+
+		if (StringUtils.hasText(this.role)) {
+			this.roleController = getBeanFactory()
+					.getBean(IntegrationContextUtils.INTEGRATION_LIFECYCLE_ROLE_CONTROLLER,
+					SmartLifecycleRoleController.class);
+
+			this.roleController.addLifecycleToRole(this.role, this);
```

Comment:
Needs null check - also, if endpoints now register themselves, shouldn't we use the parsers to use the same technique? https://github.com/spring-projects/spring-integration/pull/1483/files#diff-1fda9b32dc8bd19e9984d90ebf376ef6

Reply:
Yeah... We have to catch:
```
catch (NoSuchBeanDefinitionException e) {
		this.logger.error("No LifecycleRoleController in the context");
}
```

Well, we can't use `addLifecycleToRole()` from the XML parser since we don't have bean instances there.
The manipulation with the `BeanDefinition` attributes is correct there.
IMO...

Reply:
? Can't we just set the `role` property on the endpoint BD from the xml?